app/controllers/paginable/plans_controller.rb

```diff
@@ -42,9 +42,15 @@ def org_admin
     unless current_user.present? && current_user.can_org_admin?
       raise Pundit::NotAuthorizedError
     end
+    #check if current user if super_admin
+    @super_admin = current_user.can_super_admin?
+    plans = @super_admin ? Plan.all : current_user.org.plans
+    plans = plans.joins(:template, roles: [user: :org]).where(Role.creator_condition)
```

Comment:
you'd likely want to do this joins+where in the plans_controller as well . Also consider using `includes` if the roles are displayed in the table layout. Includes forces eager loading whereas joins implements an `INNER JOIN` on the relation and uses lazy loading

Reply:
> you'd likely want to do this joins+where in the `plans_controller` as well

The joins+where is only needed for the sort/filter options (to ensure that the plan's creator's org can be queried), so I think should only be relevant to the paginable controller.  

> consider using `includes` if the roles are displayed in the table layout

That table's calling the `plan.owner` function to get an owner's name, which does it's own queries to `Role` and `User`, so I don't think the includes would help with access in this case